File: src/python/scraper/videos/verified.py
from .base import VideoScraper, parse_response

class VerifiedScraper(VideoScraper):
    def get_verified(self, page: int = 1, type="women") -> str:
        page -= 1
        if type == "women":
            endpoint = "verified"
        elif type == "men":
            endpoint = "verified_men"
        elif type == "gay":
            endpoint = f'verified_gay'
        elif type == "trans":
            endpoint = f'verified_trans'
        else:
            raise ValueError("Endpoint must be either women, men, gay or trans")
        data = self.scrape(endpoint, {'page': page} if page else None)
        return parse_response(page, data, is_fresh=True)

File: src/api/videos/verified.ts
import VideoScraper from "api/videoScraper";
import Parser from "api/utils/parser";

class VerifiedScraper extends VideoScraper {
  private parser: Parser;
  private endpoint: string;
  constructor(isFresh: boolean = true) {
    super();
    this.parser = new Parser(isFresh);
    this.endpoint = "verified";
  }
  async getVerified(type: VerifiedType = "women", page: number = 1) {
    if (type !== "women") {
      this.endpoint += `_${type}`;
    }
    if (page <= 0) {
      this.endpoint += "_p";
    }
    const data = await this.scrape(this.endpoint, { page });
    return this.parser.parseResponse(page, data);
  }
}

export default VerifiedScraper;
